js/main.js

```diff
@@ -20,12 +23,30 @@ function logForm(event) {
     notes: $textArea.value,
     entryId: data.nextEntryId
   };
-  data.nextEntryId += 1;
-  data.entries.unshift(formObject);
-  $setImg.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $formSubmit.reset();
-  $ul.prepend(entryDOM(formObject));
-  switchViews("entries");
+  if (data.editing !== null){
+    for (var k = 0; k < data.entries.length; k++) {
+      if (data.editing === data.entries[k].entryId) {
+        var changeObject = {
+          title: $inputTitle.value,
+          photoUrl: $imgUrl.value,
+          notes: $textArea.value,
+          entryId: data.entries.length - k
+        }
+        data.entries.splice(k, 1, changeObject)
+        var $findLi = document.querySelectorAll('li')
+        $findLi[k].replaceWith(entryDOM(changeObject))
```

Comment:
This isn't the best way to handle this, your list of entries on the DOM may potentially not exactly match the list of entries stored in localStorage. You would be better of looping over the list of entries to find the one with the same `data-entry-id` as the entryId stored in `data.editing`